### ISSUE TITLE
Can't reach sorry-cypress from a docker container running on the same host with --net=host

### ISSUE THREAD

**Issue by tellyo-patryk** [sorry-cypress/sorry-cypress#145]
Hello, I've contacted the project team on slack yesterday and I was told to open an issue so here it is.

I'm running sorry-cypress on a host together with jenkins that runs the job that fires up a container that's compliant with the build environment with added cypress dependencies and cypress.

I've added this to the docker-compose.minio.yml, because otherwise my network would get broken:
```networks:
  default:
    ipam:
      driver: default
      config:
        - subnet: 172.50.0.0/16
```

I've tried to reach the sorry-cypress director from the container with both localhost:1234 and 172.50.0.5:1234 (in this case it's the container's address)

I'm running cypress inside the container like this (maybe that's the issue, I suppose):
`npm run cy:run --record --key xxx --parallel --verbose --ci-build-id $BUILD_ID`

**Comment by dragon788** [comment]
Have you defined any `ports` for the services in your file, those would define what ports would be listening on the host.

In your `cypress` container you also need to ensure you have changed the `api_url` to an appropriate value.

I use something like this as the `--entrypoint` for my `cypress/included` but it should work just as well with your custom container, with the bonus that you don't need to specify the `npm run` bit. You can override the default `http://director:1234` by passing in another value as `--env CYPRESS_API_URL='http://172.50.0.5:1234'` or similar.

```
#!/bin/bash -x

# This entrypoint file is used/required if targeting local sorry-cypress for parallelization
# Set api_url to desired host
sed -i -e 's|api_url:.*$|api_url: "'${CYPRESS_API_URL:-http://director:1234}'"|g' /*/.cache/Cypress/*/Cypress/resources/app/packages/server/config/app.yml

# Sleep briefly so starting N containers at once doesn't overwhelm the director
sleep $((($RANDOM % 10) + 3))s

# Note we don't pass the `run` of cypress run in case we wanted to use `open` for local testing from inside Docker
exec cypress "$@"
```

The `docker run` might look like this:

```
docker run --rm \
  --env CYPRESS_API_URL='http://172.50.0.5:1234' \
  --mount type=bind,source=$(pwd),target=/e2e \
  --workdir /e2e \
  --entrypoint cypress-entrypoint.sh \
  my-cypress \
  run --record --key xxx --parallel --verbose --ci-build-id $BUILD_ID
```

**Comment by agoldis** [comment]
Hy, @tellyo-patryk Please tell me if what @dragon788 suggested helped

**Comment by tellyo-patryk** [comment]
I didn't notice the comment above, I switched from my container to cypress/base:12 since mine had nodejs 10 and fired up a t3a.small for the sorry-cypress dashboard since our Jenkins was getting overcrowded. I then changed the api_url to that new machine and I found out that I need to add the machine hosting the dashboard to /etc/hosts since I couldn't figure out how to change uploadUrl, I then faced issues with loading frontend and I needed to change GRAPHQL_SCHEMA_URL to have the machine's ip in docker-compose.minio.yml, later I noticed that video doesn't load in frontend and after replacing MINIO_URL with the machine's ip everything works. Thank you for your help and for the project. I also found out that running npm run cy:run with the package installed as a dependency wouldn't work while installing cypress globally and running with cypress run works.
